packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -276,31 +367,40 @@ class SnackBar extends StatelessWidget {
             child: child,
           );
         },
-        child: snackbar,
-      ),
-    );
+        child: snackBar,
+      );
+    }
+
+    return ClipRect(child: snackBarTransition);
   }
 
   // API for Scaffold.addSnackBar():
 
-  /// Creates an animation controller useful for driving a snack bar's entrance and exit animation.
-  static AnimationController createAnimationController({ @required TickerProvider vsync }) {
+  /// Creates an animation controller useful for driving a snack bar's entrance
+  /// and exit animation.
+  static AnimationController createAnimationController({
```

Comment:
I don't think you need to reformat all this.

Reply:
Done